lms/djangoapps/courseware/date_summary.py

```diff
@@ -268,7 +268,7 @@ def date_type(self):
     @property
     def title(self):
         enrollment = CourseEnrollment.get_enrollment(self.user, self.course_id)
-        if enrollment and self.course.end and enrollment.created > self.course.end:
+        if self.course.self_paced and enrollment and self.course.start and enrollment.created > self.course.start:
```

Comment:
This will have an impact on 2 cases:

- Instructor-paced course with past dates:
  - `Course Starts` replaced by `Enrollment Date`
- Self Paced course with start date in past and end date in future:
  - 'Course Starts' replaced by `Enrollment Date`, Regardless of user enrollment. If user is not enrolled and you visit dates tab, it will display `Enrollment Date`

Reply:
This looks good. It displayed the wrong label due to cache.